packages/core/src/utils/gitIgnoreParser.ts

```diff
@@ -186,13 +122,68 @@ export class GitIgnoreParser implements GitIgnoreFilter {
         return false;
       }
 
-      return this.ig.ignores(normalizedPath);
+      const ig = ignore();
+
+      // Always ignore .git directory
+      ig.add('.git');
+
+      // Load global patterns from .git/info/exclude on first call
+      if (this.globalPatterns === undefined) {
+        const excludeFile = path.join(
+          this.projectRoot,
+          '.git',
+          'info',
+          'exclude',
+        );
+        this.globalPatterns = fs.existsSync(excludeFile)
+          ? this.loadPatternsForFile(excludeFile)
+          : [];
+      }
+      ig.add(this.globalPatterns);
+
+      const pathParts = relativePath.split(path.sep);
+
+      const dirsToVisit = [this.projectRoot];
+      let currentAbsDir = this.projectRoot;
+      // Collect all directories in the path
+      for (let i = 0; i < pathParts.length - 1; i++) {
+        currentAbsDir = path.join(currentAbsDir, pathParts[i]);
+        dirsToVisit.push(currentAbsDir);
+      }
+
+      for (const dir of dirsToVisit) {
+        const relativeDir = path.relative(this.projectRoot, dir);
```

Comment:
we might want to tweak this to directly support `includeDirectories`. For now we should probably just ignore this logic if the path.relative left the root dir.
In the future we could  figure out which of the included directories the path is from and only consider the git ignores for that included directory.